lib/lighthouse/benefits_intake/sidekiq/submission_status_job.rb

```diff
@@ -0,0 +1,191 @@
+# frozen_string_literal: true
+
+require 'lighthouse/benefits_intake/service'
+require 'dependents/benefits_intake/submission_handler'
+require 'pcpg/benefits_intake/submission_handler'
+require 'vre/benefits_intake/submission_handler'
+
+# Datadog Dashboard
+# https://vagov.ddog-gov.com/dashboard/4d8-3fn-dbp/benefits-intake-form-submission-tracking
+module BenefitsIntake
+  class SubmissionStatusJob
+    include Sidekiq::Job
+
+    sidekiq_options retry: false
+
+    STATS_KEY = 'api.benefits_intake.submission_status'
+    STALE_SLA = Settings.lighthouse.benefits_intake.report.stale_sla || 10
+    BATCH_SIZE = Settings.lighthouse.benefits_intake.report.batch_size || 1000
+
+    # any status not listed will result in 'pending'
+    STATUS_RESULT_MAP = {
+      expired: 'failure', # Indicates that documents were not successfully uploaded within the 15-minute window
+      error: 'failure',   # Indicates that there was an error. Refer to the code and detail for further information
+      vbms: 'success',    # Submission was successfully uploaded into a Veteran's eFolder within VBMS
+      success: 'pending', # Submission was successfully received into Lighthouse systems
+      pending: 'pending', # Submission is being processed
+      stale: 'stale'      # Exceeds SLA (service level agreement) days for submission completion; non-lighthouse status
+    }.freeze
+
+    # A hash mapping form IDs to their corresponding handlers.
+    # This constant is intentionally mutable.
+    # @see register_handler
+    FORM_HANDLERS = {} # rubocop:disable Style/MutableConstant
+
+    # Registers a form class with a specific form ID.
+    #
+    # @param form_id [String] The form ID to register.
+    # @param form_handler [Class] The class associated with the form ID.
+    def self.register_handler(form_id, form_handler)
+      FORM_HANDLERS[form_id] = form_handler
+    end
+
+    # Registers handlers for various form IDs.
+    {
+      '686C-674' => Dependents::BenefitsIntake::SubmissionHandler,
+      '28-8832' => PCPG::BenefitsIntake::SubmissionHandler,
+      '28-1900' => VRE::BenefitsIntake::SubmissionHandler
+    }.each do |form_id, handler_class|
+      register_handler(form_id, handler_class)
+    end
+
+    def initialize(batch_size: BATCH_SIZE)
+      @batch_size = batch_size
+    end
+
+    def perform(form_id = nil)
+      log(:info, 'started')
+
+      pending_attempts = FormSubmissionAttempt.where(aasm_state: 'pending').includes(:form_submission)
+
+      # filter running this job to a specific form_id/form_type
+      pending_attempts.select! { |pa| pa.form_submission.form_type == form_id } if form_id
+
+      batch_process(pending_attempts) unless pending_attempts.empty?
+
+      log(:info, 'ended')
+    end
+
+    private
+
+    attr_reader :batch_size, :form_id
+
+    def log(level, msg, **payload)
```

Comment:
We don't want to use our new logging logic here?

Reply:
I was going to use it, but there is no metric/stastsd to be sent in this case and the call_location is irrelevant since it is in a periodic job